WooCommerce/Classes/POS/Presentation/ItemRowView.swift

```diff
@@ -59,17 +61,24 @@ struct ItemRowView: View {
     private var productImage: some View {
         if dynamicTypeSize >= .accessibility3 {
             EmptyView()
-        } else if let imageSource = cartItem.item.productImageSource {
-            ProductImageThumbnail(productImageURL: URL(string: imageSource),
-                                  productImageSize: Constants.productCardSize,
-                                  scale: scale,
-                                  foregroundColor: .clear)
-            .frame(width: min(Constants.productCardSize * scale, Constants.maximumProductCardSize),
-                   height: Constants.productCardSize * scale)
-            .clipped()
+        } else if let imageSource = viewModel.cartItem.item.productImageSource {
+            if let hasImageInCache = viewModel.cachedImage {
+                Image(uiImage: hasImageInCache)
+                    .resizable()
+                    .scaledToFill()
+                    .frame(width: min(Constants.productCardSize * scale, Constants.maximumProductCardSize),
+                           height: Constants.productCardSize * scale)
+                    .clipped()
+            } else {
+                ProductImageThumbnail(productImageURL: URL(string: imageSource),
+                                      productImageSize: Constants.productCardSize,
+                                      scale: scale,
+                                      foregroundColor: .clear)
+                .frame(width: min(Constants.productCardSize * scale, Constants.maximumProductCardSize),
+                       height: Constants.productCardSize * scale)
+                .clipped()
```

Comment:
There's a bit of duplication here. I though of moving the logic to `ProductImageThumbnail` but that would increase the number of places changes might affect and we have to test. Perhaps a duplicate of `ProductImageThumbnail` for POS would make sense? So the `CartItemRowViewModel` can decide if the image that we load is from cache or not, rather than having this logic in the view.